toolbox/modules/enjim_pdm.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from functools import lru_cache
+from re import search
+from typing import Optional, Generator, Tuple, List, Dict
+
+from transformers import pipeline
+
+from toolbox.core.models import Episode, Turn
+from toolbox.datasets.enjim import EnjimDataset, EnjimAgent, setup_sqlite
+from toolbox.modules import BaseModule
+from toolbox.modules.registry import ModuleRegistry
+from toolbox.parsers.bb_code import BBCtoMD
+from toolbox.utils.dataset import get_data_path, get_config
+
+
+class EnjimPDM(BaseModule, metaclass=ModuleRegistry):
+    """
+    Persona Dialogue Module based on the Enjim dataset.
+    NOTE: All the summarizing stuff is just there until whatever is going to be done with vector dbs is figured out.
+    """
+    CACHE_QUERY = "SELECT summary FROM summary_cache WHERE forum_shortname = ? AND text_id = ? AND max_length = ?"
+    CACHE_INSERT = "INSERT INTO summary_cache (forum_shortname, text_id, max_length, summary) VALUES (?, ?, ?, ?)"
+
+    def __init__(self):
+        self.logger = logging.getLogger(self.__class__.__name__)
+        self.settings = get_config('enjim')
+        self.summarizer = pipeline("summarization", model="philschmid/bart-large-cnn-samsum")
+        self.cache_db = setup_sqlite(get_data_path("enjim"), EnjimDataset.CACHE_DB, self.settings['cache_db'],
+                                     logger=self.logger)
+
+    def generator(self) -> Generator[Episode, None, None]:
+        for episode in EnjimDataset():
+            thread_subject: str = episode.thread_subject
+            agents: Dict[str, EnjimAgent] = episode.agents
+            posts: List[Tuple[str, str]] = episode.posts
+            bot_name = posts[0][0]
+            turns = [Turn(utterance=spoken if not search(BBCtoMD.INVALID_RESULT, spoken)
+                     else self.summarize(spoken, self.settings['max_scenario_chars'], None, None),
+                          speaker=speaker, human_speaker=speaker != bot_name) for speaker, spoken in
+                     posts]
+            participant_personas = {ag.name: self.summarize_char(ag, self.settings['max_persona_chars'],
```

Comment:
A couple minor issues with the personas:

- Empty personas are generated sometimes, resulting in processed text that looks like:
  ```
  A's Persona: 
  B's Persona: (actual persona text here)
  Scenario: (proper scenario here)
  ... 
  ```
  Which is a little wasteful token-wise and might be confusing to the model when training.
- They seem to be unparsed query results instead of cleaned text, so instead of generating
  ```
   A's Persona: A is like this and that
  ```
  They look like:
  ```
  A's Persona: ("A is like this and that", )
  ```
  This is also happening on `world_scenario`.